REFERENCE.md

```diff
@@ -0,0 +1,156 @@
+## Reference
+
+### Classes
+
+#### Public Classes
+
+* [`conjur`](#conjur-class): Establishes a host identity on the node.
+
+### Functions
+
+#### Public Functions
+
+* [`conjur::secret`](#conjursecret): Retrieve a secret using the host identity.
+
+### Facts
+
+#### Private facts
+
+* [`conjur`](#conjur-fact): Reports current node configuration and identity.
+
+### `conjur` class
+
+This class establishes Conjur host identity on the node so that secrets can be
+fetched from Conjur. The identity and Conjur endpoint configuration can be
+pre-configured on a host using `/etc/conjur.conf` and `/etc/conjur.identity` on Linux, or
+Windows Registry and Window Credentials Manager on Windows, (by way of
+[`conjur` fact](#conjur-fact)) or provided as parameters. The identity can also be
+bootstrapped using the host factory token parameter.
+
+When this class is instantiated in such a way that it has access to an API key (`authn_api_key`),
+the identity is persisted on the agent using the same OS specific machine identity
+artifacts as mentioned in the pre-configured case above. This happens when the host
+factory parameter is used since a host's API key is retrieved as part of that flow. _Note
+that when this persistence occurs the `cert_file` parameter is always converted to the
+`ssl_certificate` equivalent for the artifact._
+
+#### Note
+
+Several parameters (ie. API keys) are of `Sensitive` data type.
+To pass a normal string, you need to wrap it using `Sensitive("example")`.
+
+#### Parameters
+
+##### `account`
+Conjur account authority name. Optional for v4, required for v5.
+
+##### `appliance_url`
+A Conjur endpoint (with trailing `/api/` for v4).
+
+##### `authn_login`
+User username or host name (prefixed with `host/`).
+
+##### `authn_api_key`
+API key for a user or host. Must be `Sensitive` if supported.
+
+##### `cert_file`
+File path to X509 certificate of the root CA of Conjur, PEM formatted. This file is read
+from the **Puppet server**. Takes precedence over `ssl_certificate`.
+
+##### `ssl_certificate`
+Content of the X509 certificate of the root CA of Conjur, PEM formatted.
+When using Puppet's `file` function, the path to the cert must be absolute.
+Overwritten by the contents read from `cert_file` when it is present.
```

Comment:
I see a _lot_ of references to cert file here. you may need a rebase and manual fixes on this for that

Reply:
yup, gonna need a rebase. will work on that now.